### DISCUSSION TITLE
How to build a convolution with channels ("colors") at each point?

### DISCUSSION THREAD

**Discussion by OlfwayAdbayIgbay** [e3nn/e3nn#418] [General]
I have point clouds of shape (N,9), where the first 3 features are explicit coordinates, and the other 6 features are "properties". How would I represent data like this in e3nn?
My understanding of how to wrap the data in torch_geometric:

```python
from torch_geometric.data import Data

def wrap_pc(pc):
  pos = pc[:,:3]
  x = pc[:,3:]
  return Data(x=x, pos=pos)
```

Now I assume my input irreps are "3x0o + 6x0e" because the coordinates are changing sign under inversion?

But how would I now build a convolution that also takes the features into account, I'm a bit lost on this. The convolution from the tutorial, as far as I can see, is purely euclidean distance based, unless I'm not getting something.

**Comment by mariogeiger** [comment]
The irreps of the `x` attribute is `6x0e` since you got rid of the positions. `.pos` are positions, when you will compute relative positions in your architecture you will be able to treat them as vector: `1o`
